test/CompanyService.Business.UnitTests/Commands/Company/CreateCompanyCommandTests.cs

```diff
@@ -1,344 +1,544 @@
+               .Returns(_company);
+
+            var ex = Assert.Throws<BadRequestException>(() => _command.Execute(_request));
+            Assert.That(ex.Message, Is.EqualTo("Company already exists"));
+
+            _autoMock.Verify<ICompanyRepository>(
+               x => x.Add(It.IsAny<DbCompany>()),
+               Times.Never());
+
+            _autoMock.Verify<ICreateCompanyRequestValidator>(
+               x => x.Validate(It.IsAny<IValidationContext>()),
+               Times.Never);
+
+            _autoMock.Verify<IDbCompanyMapper>(
+               x => x.Map(It.IsAny<CreateCompanyRequest>()),
+               Times.Never());
+
+            _autoMock.Verify<IRequestClient<IUpdateSmtpCredentialsRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+            It.IsAny<object>(), default, default).Result.Message, Times.Never);
+
+            _autoMock.Verify<IRequestClient<ICreateAdminRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+               It.IsAny<object>(), default, default).Result.Message, Times.Never);
+        }
+
+        [Test]
+        public void ShouldReturnFailedResponseWhenUpdateSMTPResponseThrowAndCatchException()
+        {
+            _autoMock
+                .Setup<IOperationResult<bool>, bool>(x => x.IsSuccess)
+                .Returns(false);
+
+            _autoMock
+                .Setup<IOperationResult<bool>, List<string>>(x => x.Errors)
+                .Returns(new List<string> { "some error" });
+
+            _autoMock.Setup<Response<IOperationResult<bool>>, IOperationResult<bool>>(x => x.Message)
+                .Returns(_autoMock.GetMock<IOperationResult<bool>>().Object);
+
+            _autoMock
+               .Setup<IRequestClient<IUpdateSmtpCredentialsRequest>, Task<Response<IOperationResult<bool>>>>(
+                x => x.GetResponse<IOperationResult<bool>>(
+                       It.IsAny<object>(), default, default))
+               .Throws(new Exception());
+
+            var expected = new OperationResultResponse<Guid>
+            {
+                Status = Kernel.Enums.OperationResultStatusType.Failed,
+                Errors = new List<string> { "Can not update smtp credentials." }
+            };
+
+            SerializerAssert.AreEqual(expected, _command.Execute(_request));
+
+            _autoMock.Verify<ICompanyRepository>(
+               x => x.Add(It.IsAny<DbCompany>()),
+               Times.Never());
+
+            _autoMock.Verify<ICreateCompanyRequestValidator>(
+               x => x.Validate(It.IsAny<IValidationContext>()),
+               Times.Once);
+
+            _autoMock.Verify<IDbCompanyMapper>(
+               x => x.Map(It.IsAny<CreateCompanyRequest>()),
+               Times.Never());
+
+            _autoMock.Verify<IRequestClient<IUpdateSmtpCredentialsRequest>>(
+               x => x.GetResponse<IOperationResult<bool>>(
+            It.IsAny<object>(), default, default).Result.Message, Times.Never);
```

Comment:
add tab

Reply:
Why times never if your method ShouldReturnFailedResponseWhenUpdateSMTPResponseThrowAndCatchException use this method

Reply:
done, because here we are checking that our method returns values. And then an exception is triggered and it does not return anything.